docs/docs/when-to-use-use.md

```diff
@@ -0,0 +1,240 @@
+
+There is a potential for a whole "usable system" here, piggy-backing off of the [helper-manager](https://rfcs.emberjs.com/id/0625-helper-managers/) pattern.
+
+</details>
+
+### `@use`
+
+
+#### "@use thisProperty = DefinedAsThis"
+
+This `@use` decorator _replaces_ the property with a getter that gets the underlying value of the resource. You'll see what happens in non-decorator form later.
+
+<details><summary>looking under the hood</summary>
+
+You could think of what the decorator does in this way:
+
+```js
+class Demo extends Component {
+  @use data = Clock;
+}
+```
+is (at runtime) transformed to:
+```js
+class Demo extends Component {
+  #data1 = invokeHelper(this, Clock);
+  
+  get data() {
+    return getValue(this.#data1);
+  }
+}
+```
+
+Note that the _actual_ return value of `resource` is an internal and private object, and interaction with that object is not supported in user-space.
+
+</details>
+
+In this example, the resource is the value of `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource(() => {
+    /* ... */
+    return 2;
+});
+
+class Demo extends Component {
+    @use data = Clock;
+
+    <template>
+        {{this.data}} prints 2
+    </template>
+}
+```
+
+In this example, the `resource` is still the value returned to `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+function LocalizedClock(locale) {
+    return resource(() => {
+        /* ... */
+        return 2;
+    });    
+}
+
+class Demo extends Component {
+    @use data = LocalizedClock('en-US');
+
+    <template>
+        {{this.data}} prints 2
+    </template>
+}
+```
+
+#### "@use(ThisResource) namedThis"
+
+This variant of `@use` _only_ works if the Resource does not take arguments.
+It is the most constrained / least-flexible way to use `@use`.
+
+<details><summary>but why?</summary>
+
+This is because the left-hand side of decorators exists outside of an instance, so no instance properties may be accessed, because there is no instance.
+
+One way to think of how decorators work right now, is to "in your head", transform the code from:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource( ... );
+
+class Demo extends Component {
+    @use(Clock) time;
+}
+```
+
+To
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource( ... );
+
+class Demo extends Component {}
+
+let decorator = use(Clock);
+Object.defineProperty(Demo.prototype, {
+    get() {
+        return decorator(this, 'data', /* ... */);
```

Comment:
would this be `'time'` instead of `'data'` (assuming this is an accidental carryover from the previous example)?

Reply:
yup! thanks!